src/main/java/com/direwolf20/buildinggadgets/common/tools/IAtopSupport.java

```diff
@@ -0,0 +1,15 @@
+package com.direwolf20.buildinggadgets.common.tools;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+
+/**
+ * Such mode should work as if placing atop without any offset.
```

Comment:
```suggestion
```
I have no idea what this description is trying to tell me. Please add some description of what an implementation is responsible for.